Kotlin 2.2 support

LouisCAD opened the pull request.

XilinJia committed:
3.1.0 commit

(cherry picked from commit 3253651133fded30dc00d70faf14f57c8e547af5)

XilinJia committed:
3.1.0 commit

(cherry picked from commit 4b32ff3cef253338dd9275e7bea73041538a285b)

XilinJia committed:
3.2.0 commit

(cherry picked from commit 8d5c4c5824ddbbcecf83a645d926893531e48daa)

XilinJia committed:
3.2.1 commit

(cherry picked from commit aa3f0d5e3b508dbd3362ffd0e36740de6166cb3c)

XilinJia committed:
3.2.2 commit

(cherry picked from commit 6d018d1c107b4423556d1afe2e66a794536b3b22)

XilinJia committed:
first rebranding from Realm Kotlin

(cherry picked from commit 94d2073df5ce181ec4413de2683a883b4fad1640)

XilinJia committed:
added local usage in readme

(cherry picked from commit 5e4e04c0fe30da5621f50ccf53a41e69ce40694a)

XilinJia committed:
added local usage in readme

(cherry picked from commit f6e873bf8e6f68b49889ad97cfaafd17d438998a)

XilinJia committed:
added local usage in readme

(cherry picked from commit 340a9d4d0801987ca2f8af29df6428be8ea82433)

XilinJia committed:
3.2.4 commit

(cherry picked from commit 6347f11ffbb90024a8ac237220148d535b734962)

XilinJia committed:
3.2.4 commit

(cherry picked from commit 9e6e7ac95974090255b0c117c52b8f827fdb6a60)

XilinJia committed:
3.2.4 commit

(cherry picked from commit 38de15823c2ade20ed547eb125e9cc8593864d9a)

XilinJia committed:
3.2.4 commit

(cherry picked from commit 4fba2bf6c1ed0bc6d50093e2a12204714fb18277)

XilinJia committed:
3.2.5 commit

(cherry picked from commit 4c001eca51a8fd42da99973f5c5dbc2c9b6335de)

XilinJia committed:
3.2.5 commit

(cherry picked from commit 38c68245d46438cb9f31b0428f3baee33a8c458e)

XilinJia committed:
3.2.6 commit

(cherry picked from commit 325808eaba1b09f1e426053615208ccce5d4ec76)

XilinJia committed:
3.2.6 commit

(cherry picked from commit 1cc034f865e6e84eb5a7ab84dcd78c1eff06278a)

XilinJia committed:
3.2.6 commit

(cherry picked from commit bc32abffdcc5c3763be126c2a86b66e227305eca)

tinoue committed:
fix build break on macOS

(cherry picked from commit adf724a526e54321b9159202884b328060233c71)

XilinJia committed:
3.2.7 commit

(cherry picked from commit 09af8b63d9abdfe8ce8272e04f36e56b6bcb6ef2)

XilinJia committed:
3.2.7 revised

(cherry picked from commit aea2456eea256537484a4a2dc918932df9908951)

tinoue committed:
Update README to reflect testing status for iOS, and MacOS

(cherry picked from commit e38ff863c09c7bb99c1ab92cb318344a96308af5)

XilinJia committed:
preliminary Kotlin 2.2.0 build

(cherry picked from commit 0b1d4f2d6c55d9a848cce4488c414088e36aa538)

XilinJia committed:
kotlin 2.2.0 RC

(cherry picked from commit d9abc67498216e59a785e296dba186b8db0e234e)

XilinJia committed:
update readme

(cherry picked from commit 8de9e0ee357b3cab0de819afc392277a4acffee1)

XilinJia committed:
3.2.8 RC1

(cherry picked from commit da449255649a8ae21eb18a0b87fa2f8afbbc6e82)

LouisCAD committed:
Fix renaming script

LouisCAD committed:
Rename krdb back to realm-kotlin

LouisCAD committed:
Also rename the Gradle project

LouisCAD committed:
Remove unwanted check-pr-title.yml GitHub workflow

LouisCAD committed:
Rename io.github.xilinjia.krdb packages back to io.realm.kotlin

LouisCAD committed:
Switch order of Gradle plugins declaration

LouisCAD committed:
Update README.md and CONTRIBUTING.md

LouisCAD committed:
Remove unneeded krdb-added widgets.qss file

LouisCAD committed:
Update SCM config

LouisCAD reviewed (commented):
Self-reviewed everything.

LouisCAD committed:
Add troubleshooting for Gradle sync in CONTRIBUTING.md

LouisCAD committed:
Workaround reflection-related performance issue

This should fix https://github.com/realm/realm-kotlin/issues/1544

See this PR that was the inspiration: https://github.com/realm/realm-kotlin/pull/1851

LouisCAD committed:
kotlinx.coroutines 1.10.2

LouisCAD committed:
Change plugin id and maven coordinates of artifacts

Changed them from io.realm.kotlin to com.infomaniak.realm.kotlin prior to publication

LouisCAD committed:
Generate plugin marker artifact by default

LouisCAD committed:
Update dependency notations and plugin id

LouisCAD committed:
Update project URLs

LouisCAD committed:
Replace jcenter with mavenCentral in root project buildscript repos

LouisCAD committed:
Remove redundant publication

It's already set by the plugin "com.gradle.plugin-publish"

LouisCAD committed:
Remove the publication of dokka jars

This should make the build a bit faster, and the output lighter

LouisCAD committed:
Update Maven Central publishing setup

Replace legacy gradle-nexus plugin with GradleUp/nmcp
that will work with the new central.sonatype.com portal

LouisCAD committed:
Update troubleshooting info in CONTRIBUTING.md

LouisCAD committed:
Workaround Gradle issue #26091

https://github.com/gradle/gradle/issues/26091#issuecomment-1722947958

LouisCAD committed:
Update GPG-related key names

LouisCAD committed:
Update heading in CONTRIBUTING.md

sirambd reviewed (commented):
Approved all files until 5fbf32d

LouisCAD committed:
fix: Update internal-ish plugin id

The build of consumers would fail because the gradle plugin
was adding a dependency to another plugin with the old
maven coordinates that we cannot publish to.

"io.realm.kotlin" -> "com.infomaniak.realm.kotlin"

LouisCAD committed:
3.2.8-1 commit

LouisCAD committed:
Make publishing automatic

sirambd commented on buildSrc/src/main/kotlin/Config.kt:
```diff
@@ -80,64 +80,66 @@ object Realm {
-        const val organization = "MongoDB"
-        const val organizationUrl = "https://www.mongodb.com"
+        const val name = "Infomaniak"
+        const val email = "louis.cognault@infomaniak.com"
```
Replace with `mobile+libraries@infomaniak-dev.ch`

LouisCAD committed:
Update developer email put in the maven pom file
